### DISCUSSION TITLE
How to check if the test run status was failed or succeeded?

### DISCUSSION THREAD

**Discussion by makhaly** [xunit/xunit#2622] [Question] [answered]
How to check if the test method status was failed or succeeded?
I am using xUnit with Playwright to test one of our applications, I need to enable tracing of Playwright only in the case of a test failure.
If I can get the test Context and check if the last test was failed, I will be able to save the tracing files and get to know the problem.

**Comment by bradwilson** [comment, marked as answer]
There is no context object available in v2, so this isn't possible today.

**Reply by bradwilson** [reply]
FYI, there will be a context object available in v3.

**Reply by makhaly** [reply]
Great News. When will the v3 be available?

**Reply by mrobinson-campion** [reply]
In my project I'd created `BasePageTest` which inherits from `PageTest`. To differentiate between passing and failing tests in `DisposeAsync`, I use `TestOk` which is provided by `Microsoft.Playwright.Xunit.ExceptionCapturer`.

**Reply by bradwilson** [reply]
In v3, you can use `TestContext.Current.TestState.Result` and compare it against `TestResult.Passed`, `TestResult.Failed`, `TestResult.Skipped`, or `TestResult.NotRun`. There are many other pieces of information available in `TestContext.Current.TestState` (for example, getting exception information for a failed test).

**Reply by mrobinson-campion** [reply]
Playwright 1.55 is available, with support for Xunit v3.

**Comment by mrobinson-campion** [comment]
It looks like xUnit.v3 support is being added to the next version of Playwright:
https://github.com/microsoft/playwright-dotnet/pull/3097
